card/src/main/java/com/adyen/checkout/card/DefaultCardDelegate.kt

```diff
@@ -310,9 +310,19 @@ internal class DefaultCardDelegate(
             isDualBranded = isDualBrandedFlow(filteredDetectedCardTypes),
             kcpBirthDateOrTaxNumberHint = getKcpBirthDateOrTaxNumberHint(inputData.kcpBirthDateOrTaxNumber),
             componentMode = ComponentMode.DEFAULT,
+            isCardListVisible = getCardListVisibility(getCardBrands(detectedCardTypes), filteredDetectedCardTypes)
         )
     }
 
+    private fun getCardListVisibility(
+        cardBrands: List<CardListItem>,
+        detectedCardTypes: List<DetectedCardType>
+    ): Boolean = if (cardBrands.isEmpty()) {
```

Comment:
Could this be ` return cardBrands.isNotEmpty() && detectedCardTypes.isEmpty()`  instead?